### PR TITLE
Add ScrollView Indicator Overlay style

### PR DESCRIPTION
**Pull request by shwanton**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

NSScrollView can have overlay style explicitly set.
https://developer.apple.com/documentation/appkit/nsscrollerstyle/nsscrollerstyleoverlay

Overlay style will **not** take up any space in the ContentView so content can be flush with insets.

Default uses the legacy style, not the overlay.

## Changelog

[macOS] [ADDED] - Add ScrollView Indicator Overlay style 

## Test Plan

Overlay style can be enable/disabled 

https://user-images.githubusercontent.com/96719/213046667-0b09b0f9-a0ad-4758-9ad8-7dfefa729d71.mp4

### CODE REVIEW

**Comment by Saadnajmi** [comment]
This PR would allow you to override the system preference and set your own default? @chiuam sounds useful 😅

File: Libraries/Components/ScrollView/ScrollView.js [hunk 1]
```diff
@@ -480,6 +480,12 @@ export type Props = $ReadOnly<{|
    * instead of vertically in a column. The default value is false.
    */
   horizontal?: ?boolean,
+  /**
+   * When true, the scroll view's inidicator style is an overlay
+   * that does not take up content view space
+   * https://developer.apple.com/documentation/appkit/nsscrollerstyle/nsscrollerstyleoverlay
+   */
+  hasOverlayStyleIndicator?: ?boolean,
```
Comment (Saadnajmi) [inline Libraries/Components/ScrollView/ScrollView.js:488]: I would add a `@platform macos` annotation here

File: React/Views/ScrollView/RCTScrollViewManager.m [hunk 2]
```diff
@@ -102,6 +102,7 @@ - (RCTPlatformView *)view // [macOS]
 RCT_EXPORT_OSX_VIEW_PROPERTY(onInvertedDidChange, RCTDirectEventBlock) // [macOS]
 RCT_EXPORT_OSX_VIEW_PROPERTY(onPreferredScrollerStyleDidChange, RCTDirectEventBlock) // [macOS]
 RCT_EXPORT_VIEW_PROPERTY(inverted, BOOL)
+RCT_EXPORT_VIEW_PROPERTY(hasOverlayStyleIndicator, BOOL)
```
Comment (Saadnajmi) [inline React/Views/ScrollView/RCTScrollViewManager.m:105]: I would use `RCT_EXPORT_OSX_VIEW_PROPERTY` so this isn't set for iOS

File: packages/rn-tester/js/examples/ScrollView/ScrollViewExample.js [hunk 3]
```diff
@@ -500,6 +500,14 @@ if (Platform.OS === 'macos') {
...
       return <InvertedContentExample />;
     },
+  },
+  {
+    title: '<ScrollView> (hasOverlayStyleIndicator = true/false)\n',
+    description:
+      "You can display <ScrollView>'s indicator using overlay style",
+    render: function (): React.Node {
+      return <ScrollIndicatorOverlayExample />;
+    },
```
Comment (Saadnajmi) [inline packages/rn-tester/js/examples/ScrollView/ScrollViewExample.js:512]: Enclosing diff tags seem off?

Comment (shwanton) [reply packages/rn-tester/js/examples/ScrollView/ScrollViewExample.js]: Lintet fixed it!

**Comment by shwanton** [comment]
> This PR would allow you to override the system preference and set your own default? @chiuam sounds useful 😅

This does allow user to override the system preference and always have overlay style enabled.

https://user-images.githubusercontent.com/96719/213052899-fa625570-eaf5-4482-9f86-5ed8889dbbdb.mp4

**Review by chiuam — APPROVED** [review]

**Pull request merged**